runtime/src/main/java/org/corfudb/runtime/collections/WrappedRocksIterator.java

```diff
@@ -0,0 +1,143 @@
+package org.corfudb.runtime.collections;
+
+import org.corfudb.runtime.exceptions.unrecoverable.UnrecoverableCorfuError;
+import org.rocksdb.RocksDBException;
+import org.rocksdb.RocksIterator;
+import org.rocksdb.RocksIteratorInterface;
+
+/**
+ *
+ * A wrapper class that makes a access to the RocksIterator safe: the rocks library does some checks but they
+ * are enforced via assert, but assert checking is disabled on many jvms by default, hence the explicit checking.
+ *
+ * Created by Maithem on 1/24/20.
+ */
+public class WrappedRocksIterator implements RocksIteratorInterface {
+
+    private final RocksIterator iterator;
+    public WrappedRocksIterator(RocksIterator iterator) {
+        this.iterator = iterator;
+    }
+
+    /**
+     * A helper method that is called before accessing the iterator.
+     * This is necessary to make sure that access happens on an owned handle.
+     */
+    private void checkHandle() {
+        if (!iterator.isOwningHandle()) {
+            throw new IllegalStateException("RocksIterator handle is not valid");
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isValid() {
+        checkHandle();
+        boolean res =  iterator.isValid();
+        checkStatus();
+        return res;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void seekToFirst() {
+        checkHandle();
+        iterator.seekToFirst();
+        checkStatus();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void seekToLast() {
+        throw new UnsupportedOperationException("seekToLast");
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void seek(byte[] target) {
+        throw new UnsupportedOperationException("seek");
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void seekForPrev(byte[] target){
+        throw new UnsupportedOperationException("seekForPrev");
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void next() {
+       checkHandle();
+       iterator.next();
+       checkStatus();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void prev() {
+        throw new UnsupportedOperationException("prev");
+    }
+
+    /**
+     * Returns the key of the current position of the iterator.
+     */
+    public byte[] key() {
+        checkHandle();
+        byte[] res = iterator.key();
+        checkStatus();
+        return res;
+    }
+
+    /**
+     * Returns the value of the current position of the iterator.
+     */
+    public byte[] value() {
+        checkHandle();
+        byte[] res = iterator.value();
+        checkStatus();
+        return res;
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void status() throws RocksDBException {
+        throw new UnsupportedOperationException("status");
```

Comment:
`checkStatus()` and `status()` should behave the same. The implementation of this interface can be less restrictive, thus you are not required to throw `RocksDBException`, allowing you to merge these two methods.